api/nsadm/service.go

```diff
@@ -173,3 +175,17 @@ func (s *service) GetDataUserSecurity(ctx context.Context, tenant string) (bool,
 	}
 	return false, ErrUnauthorized
 }
+
+func (s *service) ListTokens(ctx context.Context, namespace string) ([]models.Token, error) {
+	ns, _ := s.store.GetNamespace(ctx, namespace)
+	if ns != nil {
+		tokens := []models.Token{}
+		for _, token := range ns.Token.(primitive.A) {
+			tokens = append(tokens, token.(models.Token))
+		}
+		return tokens, nil
+	}
+	return []models.Token{}, ErrTokenNotFound
+}
```

Comment:
Please, use the _early return_ pattern for any new code. We'll later rework existing code to avoid this wrong code pattern to make the code worse.

This could be reworked:
```go
func (s *service) ListTokens(ctx context.Context, namespace string) ([]models.Token, error) {
	ns, err := s.store.GetNamespace(ctx, namespace)
	if err != nil {
		return nil, ErrTokenNotFound
	}

	tokens := []models.Token{}
	for _, token := range ns.Token.(primitive.A) {
		tokens = append(tokens, token.(models.Token))
	}

	return tokens, nil
}
```

Reply:
Also, @gustavosbarreto mentioned we ought to avoid using MongoDB's primitives here. We ought to use it from the store API. So please take a look at this code so we start correctly.